cmd/osbuild-service-maintenance/aws.go

```diff
@@ -84,13 +99,58 @@ func AWSCleanup(maxConcurrentRequests int, dryRun bool, accessKeyID, accessKey s
 		wg.Wait()
 	}
 
+	// using err to collect both errors as we want to
+	// continue execution if one cleanup fails
+	err = nil
+	errSecureInstances := terminateOrphanedSecureInstances(a, dryRun)
+	// keep going with other cleanup even on error
+	if errSecureInstances != nil {
+		logrus.Errorf("Error in terminating secure instances: %v, continuing other cleanup.", errSecureInstances)
+		err = errSecureInstances
+	}
+
+	errSecurityGroups := searchSGAndCleanup(ctx, a, dryRun)
+	if errSecurityGroups != nil {
+		logrus.Errorf("Error in cleaning up security groups: %v", errSecurityGroups)
+		if err != nil {
+			err = fmt.Errorf("Multiple errors while processing AWSCleanup: %w and %w.", err, errSecurityGroups)
```

Comment:
Is this a valid pattern, to try to continue cleanup and collect the errors?

Reply:
I think generally error wrapping is used to wrap deeper errors, not really make a list of them. So maybe we should just add the errors to a slice and then join them together in the end? I think it might make it more readable.

Reply:
please check